ios/Classes/SwiftTruecallerSdkPlugin.swift

```diff
@@ -1,14 +1,194 @@
+    }
+    
+    private func addMainChannel(registrar: FlutterPluginRegistrar) {
+        mainChannel = FlutterMethodChannel(name: "tc_method_channel",
+                                           binaryMessenger: registrar.messenger())
+        guard let mainChannel = mainChannel else {
+            return
+        }
+        registrar.addMethodCallDelegate(self, channel: mainChannel)
+    }
+    
+    private func addEventChannel(registrar: FlutterPluginRegistrar) {
+        eventChannel = FlutterEventChannel(name: "tc_event_channel",
+                                           binaryMessenger: registrar.messenger())
+        guard let eventChannel = eventChannel else {
+            return
+        }
+        eventChannel.setStreamHandler(self)
+    }
+    
+    public func handle(_ call: FlutterMethodCall,
+                       result: @escaping FlutterResult) {
+        let method = MethodCalls(rawValue: call.method)
+        switch method {
+        case .initiateSDK:
+            trueSdk.delegate = self
+            result(true)
+        case .isUsable:
+            result(trueSdk.isSupported())
+        case .setDarkTheme,
+                .setLocale,
+                .requestVerification,
+                .verifyOtp,
+                .verifyMissedCall:
+            result("Method not implemented")
+        case .getProfile:
+            trueSdk.requestTrueProfile()
+        case .none:
+            result("Method not implemented")
+        }
+    }
+}
+
+// MARK: - App delegate methods -
+
+extension SwiftTruecallerSdkPlugin {
+    public func application(_ application: UIApplication,
+                            continue userActivity: NSUserActivity,
+                            restorationHandler: @escaping ([Any]) -> Void) -> Bool {
+        return TCTrueSDK.sharedManager().application(application,
+                                                     continue: userActivity,
+                                                     restorationHandler: restorationHandler as? ([Any]?) -> Void)
+    }
+    
+    public func application(_ app: UIApplication,
+                            open url: URL,
+                            options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool {
+        return TCTrueSDK.sharedManager().continue(withUrlScheme: url)
+    }
+}
+
+// MARK: - FlutterStreamHandler -
+
+extension SwiftTruecallerSdkPlugin: FlutterStreamHandler {
+    public func onListen(withArguments arguments: Any?,
+                         eventSink events: @escaping FlutterEventSink) -> FlutterError? {
+        self.eventSink = events
+        return nil
+    }
+    
+    public func onCancel(withArguments arguments: Any?) -> FlutterError? {
+        self.eventSink = nil
+        return nil
+    }
+}
+
+// MARK: - TCTrueSDKDelegate -
+
+extension SwiftTruecallerSdkPlugin: TCTrueSDKDelegate {
+    public func didReceive(_ profile: TCTrueProfile) {
+        var map = [String: Any]()
+        map["result"] = "success"
```

Comment:
String constants like "result", "success",  "failure", etc can be moved to a different file. 
Check `Constants.kt` in Android for reference.

Reply:
Done